core/src/main/python/discover.py

```diff
@@ -94,35 +94,52 @@ def __process_args(args):
 
     __wlst_mode = cla_helper.process_online_args(argument_map)
     target_configuration_helper.process_target_arguments(argument_map)
+    __process_model_archive_args(argument_map)
     __process_archive_filename_arg(argument_map)
     __process_variable_filename_arg(argument_map)
     __process_java_home(argument_map)
 
     return model_context_helper.create_context(_program_name, argument_map)
 
 
-def __process_archive_filename_arg(required_arg_map):
+def __process_model_archive_args(argument_map):
+    """
+    Verify that model file and/or archive file is in the argument map
+    :param argument_map: containing the CLA arguments
+    """
+    _method_name = '__process_model_archive_args'
+    if CommandLineArgUtil.ARCHIVE_FILE_SWITCH not in argument_map and \
+            CommandLineArgUtil.MODEL_FILE_SWITCH not in argument_map:
+        ex = exception_helper.create_cla_exception('WLSDPLY-06028')
+        __logger.throwing(ex, class_name=_class_name, method_name=_method_name)
+        raise ex
+
+
+def __process_archive_filename_arg(optional_arg_map):
     """
     Validate the archive file name and load the archive file object.
-    :param required_arg_map: the required arguments map
```

Comment:
Very minor, but the map passed in includes required and optional arguments after CLA processing, maybe call it `argument_map` like above method.